packages/frontend/src/app/modules/users/components/login/login.component.ts

```diff
@@ -189,9 +189,11 @@ export class LoginComponent implements OnInit, OnDestroy {
         (typeof this.returnUrl === "string" &&
           !/^\/[a-z][a-z]$/.test(this.returnUrl)))
     ) {
-      setTimeout(() => {
-        this.router.navigateByUrl(this.returnUrl as string);
-      }, 100);
+      let target = this.returnUrl as string;
```

Comment:
Ca m'étonnerait que returnUrl puisse être une Url. C'est un seul param, qui est forcément string ou undefined. 

Si "returnUrl" n'existe pas en params, on va simplement rediriger vers "this.currentLanguageService.routePrefix" 

Faudrait distinguer la logique entre rediriger vers returnUrl, ou une valeure par défaut, ça rendra le code beaucoup plus lisible, ça retire les test regex au dessus.